src/Workspaces/CoreTest/FindAllDeclarationsTests.TestSolutionsAndProject.cs

```diff
@@ -7,13 +7,32 @@
 using System;
 using System.Collections.Generic;
 using System.Linq;
+using Microsoft.CodeAnalysis.Remote;
+using Microsoft.CodeAnalysis.Remote.Testing;
+using Microsoft.CodeAnalysis.Test.Utilities;
 using Microsoft.CodeAnalysis.Text;
+using Roslyn.Test.Utilities;
 using Xunit;
+using Xunit.Abstractions;
 
 namespace Microsoft.CodeAnalysis.UnitTests
 {
     public partial class FindAllDeclarationsTests
     {
+        private readonly ITestOutputHelper _logger;
+        private AdhocWorkspace _lazyWorkspace;
+
+        public FindAllDeclarationsTests(ITestOutputHelper logger)
+        {
+            _logger = logger;
+        }
+
+        public override void Dispose()
+        {
+            base.Dispose();
+            _lazyWorkspace?.Dispose();
```

Comment:
❗ `Dispose` runs _after_ `UseExportProviderAttribute.After`. It cannot be used for disposing a test workspace without triggering a massive test performance regression.

Reply:
:(

Reply:
Sdould we add `IBeforeExportProviderCleanupDisposable.Dispose` and call that from `UseExportProviderAttribute.After`?

Reply:
That won't work - we don't have access to the test instance. Sad.

Reply:
Best way to handle it is return a `Workspace` instead of a `Solution`, and have the caller use a `using` statement.

Reply:
Yes, done.